snapcraft/commands/validation_sets.py

```diff
@@ -127,32 +132,43 @@ def run(self, parsed_args: "argparse.Namespace"):
 
 
 def _submit_validation_set(
-    edited_validation_sets: Dict[str, Any],
+    edited_validation_sets: validation_sets.EditableBuildAssertion,
     key_name: Optional[str],
     store_client: StoreClientCLI,
 ) -> None:
+    emit.debug(f"Posting assertion to build: {edited_validation_sets.json()}")
     build_assertion = store_client.post_validation_sets_build_assertion(
-        validation_sets=edited_validation_sets
+        validation_sets=edited_validation_sets.marshal()
     )
-    signed_validation_sets = _sign_assertion(
-        build_assertion.marshal(), key_name=key_name
+    build_assertion_dict = build_assertion.marshal_as_str()
```

Comment:
It sounds a bit confusing that something marshaled "as str" ends up being a dict.

Reply:
Renamed